src/Avalonia.Controls/Presenters/ScrollContentPresenter.cs

```diff
@@ -323,6 +323,7 @@ internal void AttachToScrollViewer()
             }
 
             _ownerSubscriptions?.Dispose();
+            _owner = owner;
```

Comment:
This line being moved is the actual fix.

Reply:
Yeah the side effect is that it pushes the `ScrollViewer.Content` to the SCP, which causes the SCP to read the `ILogicalScrollable` properties, which need to then be synced with the `ScrollViewer`.

Reply:
Perhaps a comment in the code about the rationale would be helpful for future eyes on this.